src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,100 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    static final int ARRAY_MIN_SIZE = 10;
+
+    private T[] array = (T[]) new Object[ARRAY_MIN_SIZE];
+    private int size;
+
     @Override
     public void add(T value) {
-
+        if (size >= array.length) {
+            array = (T[]) grow();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index >= 0 && index <= size) {
+            if (size >= array.length) {
+                array = (T[]) grow();
+            }
+            System.arraycopy(array, index, array, index + 1, size - index);
+            array[index] = value;
+            size++;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("This index is missing");
```

Comment:
Not informative message
```suggestion
            throw new ArrayListIndexOutOfBoundsException("Invalid index: " + index);
```